File: app/src/main/kotlin/com/spacex/ui/welcome/WelcomePresenter.java
package com.spacex.ui.welcome;

import androidx.annotation.NonNull;
import androidx.lifecycle.Lifecycle.Event;
import androidx.lifecycle.LifecycleObserver;
import androidx.lifecycle.LifecycleOwner;
import androidx.lifecycle.OnLifecycleEvent;
import com.spacex.ui.InteractionsHistory;
import com.spacex.ui.welcome.WelcomeContract.Presenter;
import com.spacex.ui.welcome.WelcomeContract.View;

public class WelcomePresenter implements Presenter, LifecycleObserver {

  private final WelcomeContract.View view;

  WelcomePresenter(@NonNull final View view) {
    this.view = view;
    if (view instanceof LifecycleOwner) {
      ((LifecycleOwner) view).getLifecycle().addObserver(this);
    }
  }

  @Override
  @OnLifecycleEvent(Event.ON_START)
  public void onStart() {
    view.setUpViews();
  }

  @Override
  public void onButtonClick() {
    view.navigateToRocketList();
    InteractionsHistory.setWelcomeAsShown();
  }
}

File: app/src/main/kotlin/com/spacex/ui/welcome/WelcomePresenter.kt
package com.spacex.ui.welcome

import androidx.lifecycle.Lifecycle.Event
import androidx.lifecycle.LifecycleObserver
import androidx.lifecycle.LifecycleOwner
import androidx.lifecycle.OnLifecycleEvent
import com.spacex.ui.InteractionsHistory
import com.spacex.ui.welcome.WelcomeContract.Presenter
import com.spacex.ui.welcome.WelcomeContract.View

internal class WelcomePresenter(private val view: View) : Presenter, LifecycleObserver {

    init {
        if (view is LifecycleOwner) {
            (view as LifecycleOwner).lifecycle.addObserver(this)
        }
    }

    @OnLifecycleEvent(Event.ON_START)
    override fun onStart() {
        view.setUpViews()
    }

    override fun onButtonClick() {
        view.navigateToRocketList()
        InteractionsHistory.setWelcomeAsShown()
    }
}
